### PR TITLE
Update C# generation tests for changes from HM type inference

### PR DESCRIPTION
**Pull request by samarsha**
This PR updates C# generation tests to match changes in the syntax tree related to identifier type arguments caused by microsoft/qsharp-compiler#940.

### CODE REVIEW

File: src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs [hunk 1]
```diff
@@ -2183,32 +2183,48 @@ namespace N1
                     (__arg19__.Item1, __arg19__.Item2, (1L, Result.One))
                 ))
                 .Apply((0L, Result.Zero));"
-            "partialGeneric1__.Partial((0L, _, (1L, _))).Apply((Result.Zero, Result.One));"
-            "partialGeneric2__.Partial((0L, Result.Zero, (_, Result.One))).Apply(1L);"
-            "partialGeneric2__.Partial((_, _, (1L, Result.One))).Apply((0L, Result.Zero));"
-            "partialGeneric2__.Partial((0L, _, (1L, _))).Apply((Result.Zero, Result.One));"
+            "partialGeneric1__
```
Comment (bettinaheim) [inline src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs:2186]: How come the generated code for these changed? I don't think the type inference change should have an impact on the generated C# code, or change the representation in the syntax tree.

Comment (samarsha) [reply src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs:2186]: The syntax tree did change, in that resolved identifiers now always have a type argument list attached to them, even if there was no `<T1, T2, ...>` list in the original source code.

**Review by swernli — APPROVED** [review]

File: src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs (hunk 1 above)
Comment (bettinaheim) [reply src/Simulation/CSharpGeneration.Tests/SimulationCodeTests.fs:2238]: Ok. Double check if there are any perf implications, but otherwise that's fine.

**Review by bettinaheim — APPROVED** [review]
Please quickly check the perf benchmarks before merging.

**Pull request merged**